workflows/virology/pox-virus-amplicon/README.md

```diff
@@ -0,0 +1,9 @@
+# Pox Virus Illumina Amplicon Workflow for half-genomes sequencing data
+
+This workflow generates consensus sequences from Illumina PE-sequenced ARTIC data of pox virus samples.
+
+It requires that all samples have been sequenced in two halves in two separate sequencing runs, and utilizes this property to resolve the inverted terminal repeat (ITR) sequences of pox virus genomes.
+
+The workflow uses BWA-MEM for mapping the reads from each half-genome sequencing run to a correspondingly masked version of the reference genome, merges the resulting two read mappings, and uses iVar for primer trimming and consensus sequence generation.
```

Comment:
I'm curious why you've not opted for bwa-mem2 ? Totally fine, but I thought that was a faster drop-in replacement

Reply:
That's a really good question. As so often, the honest answer is that I didn't reevaluate conclusions drawn much earlier, but your comment prompted me to start this now. So, some current thoughts and data:
- my previous (but never experimentally verified) idea was that for small viral genomes the speed gain to be expected for bwa-mem2 would be on the very lower end of what's being reported for the human genome case.
  That's because the SA is much smaller to begin with and, typically, there should be less Smith-Waterman alignments required per read (because fewer candidate seed regions should exist in a small genome) - so optimizations in both areas would matter less. otoh, at the time, I was worried that there might still be bugs waiting to be discovered in bwa-mem2.
- now, that the bwa-mem2 code base has been stable for quite some time, it's a good time to evaluate the performance instead of just speculating
- from a limited number of runs (~10) of both tools on .eu so far it seems that:
  * the `main_mem` function of bwa-mem takes ~ 25% longer than bwa-mem2's (using AVX2 mode)
  * total wall run time of bwa-mem is, on average, ~30 % longer, but that's quite variable and possibly confounded by the fact that the bwa-mem wrapper uses samtools 1.6 for sorting, while bwa-mem2 uses 1.13 (in fact, only 50% of total run time is spent in `main_mem` for each tool and final bam sizes are a tiny bit larger for bwa-mem2)
  => so, yes, the speed gain is really at the lower edge of what's reported
- at the same time, bwa-mem2 jobs still require a bit more memory than bwa-mem ones (something like 50% more on average).

All in all, I would say using bwa-mem here like in other virological iwc WFs is not something that's a  dramatically wrong decision. However, we could consider to expand the testing to more datsets and job runs, and, if the outcome is convincing, switch all relevant WFs to bwa-mem2 at some point.

Reply:
ah, and to put all those relative values into context we are talking about absolute values like:
- 15 (bwa-mem2) vs 19 secs spent in `main_mem`
- 30 vs 40 secs total job wall time
- 1 vs 1.5 GB memory

Reply:
Awesome, thanks for checking!